lib/Parse/Lexer.cpp

```diff
@@ -1558,13 +1673,37 @@ void Lexer::tryLexEditorPlaceholder() {
 }
 
 StringRef Lexer::getEncodedStringSegment(StringRef Bytes,
-                                         SmallVectorImpl<char> &TempString) {
+                                         SmallVectorImpl<char> &TempString,
+                                         unsigned Modifiers,
+                                         const std::string &ToReplace) {
+
+  // The next section of code is somewhat overcooked. It is a global replace of
+  // the string returned by getTrailingIndentOrWindowsLineEnding(token) which
+  // strips any indent that corresponds to the indent of the multiline string
+  // terminating line and normalizes line endings in the source to \n. It also
+  // removes any intial (and final) empty line.
+
+  std::string IndentStripped;
+
+  if ((Modifiers & StringLiteralMultiline) && !ToReplace.empty()) {
+    IndentStripped = Bytes;
+    size_t pos = 0;
+    while ((pos = IndentStripped.find(ToReplace, pos)) != std::string::npos) {
+      bool removeNewLine = (pos == 0 && (Modifiers & StringLiteralFirstSegment)) ||
+            ((Modifiers & StringLiteralLastSegment) &&
+             pos + ToReplace.size() == IndentStripped.size());
```

Comment:
One more nit: can these two lines be aligned to something? I believe it's the equals sign above that people would generally use.